src/protocolcodes.h

```diff
@@ -0,0 +1,335 @@
+	GAME_SERVER_PARTY_ANALYZER = 43,
+
+	// all in game opcodes must be greater than 50
+	GAME_SERVER_FIRST_GAME_OPCODE = 50,
+
+	// otclient ONLY
+	GAME_SERVER_EXTENDED_OPCODE = 50,
+
+	// NOTE: add any custom opcodes in this range (51 - 99)
+	GAME_SERVER_CHANGE_MAP_AWARE_RANGE = 51,
+	GAME_SERVER_ATTCHED_EFFECT = 52,
+	GAME_SERVER_DETACH_EFFECT = 53,
+	GAME_SERVER_CREATURE_SHADER = 54,
+	GAME_SERVER_MAP_SHADER = 55,
+	GAME_SERVER_FLOOR_DESCRIPTION = 75,
+
+	// official client ONLY
+	GAME_SERVER_PASSIVE_COOLDOWN = 94,
+	GAME_SERVER_BOSSTIARY_DATA = 97,
+	GAME_SERVER_BOSSTIARY_SLOTS = 98,
+	GAME_SERVER_SEND_CLIENT_CHECK = 99,
+	GAME_SERVER_FULL_MAP = 100,
+	GAME_SERVER_MAP_TOP_ROW = 101,
+	GAME_SERVER_MAP_RIGHT_ROW = 102,
+	GAME_SERVER_MAP_BOTTOM_ROW = 103,
+	GAME_SERVER_MAP_LEFT_ROW = 104,
+	GAME_SERVER_UPDATE_TILE = 105,
+	GAME_SERVER_CREATE_ON_MAP = 106,
+	GAME_SERVER_CHANGE_ON_MAP = 107,
+	GAME_SERVER_DELETE_ON_MAP = 108,
+	GAME_SERVER_MOVE_CREATURE = 109,
+	GAME_SERVER_OPEN_CONTAINER = 110,
+	GAME_SERVER_CLOSE_CONTAINER = 111,
+	GAME_SERVER_CREATE_CONTAINER = 112,
+	GAME_SERVER_CHANGE_IN_CONTAINER = 113,
+	GAME_SERVER_DELETE_IN_CONTAINER = 114,
+	GAME_SERVER_SET_INVENTORY = 120,
+	GAME_SERVER_DELETE_INVENTORY = 121,
+	GAME_SERVER_OPEN_NPC_TRADE = 122,
+	GAME_SERVER_PLAYER_GOODS = 123,
+	GAME_SERVER_CLOSE_NPC_TRADE = 124,
+	GAME_SERVER_OWN_TRADE = 125,
+	GAME_SERVER_COUNTER_TRADE = 126,
+	GAME_SERVER_CLOSE_TRADE = 127,
+	GAME_SERVER_AMBIENT = 130,
+	GAME_SERVER_GRAPHICAL_EFFECT = 131,
+	GAME_SERVER_TEXT_EFFECT = 132,
+	GAME_SERVER_MISSLE_EFFECT = 133,
+	GAME_SERVER_ITEM_CLASSES = 134,
+	GAME_SERVER_TRAPPERS = 135,
+	GAME_SERVER_CREATURE_HEALTH = 140,
+	GAME_SERVER_CREATURE_LIGHT = 141,
+	GAME_SERVER_CREATURE_OUTFIT = 142,
+	GAME_SERVER_CREATURE_SPEED = 143,
+	GAME_SERVER_CREATURE_SKULL = 144,
+	GAME_SERVER_CREATURE_PARTY = 145,
+	GAME_SERVER_CREATURE_UNPASS = 146,
+	GAME_SERVER_CREATURE_MARKS = 147,
+	GAME_SERVER_PLAYER_HELPERS = 148,
+	GAME_SERVER_CREATURE_TYPE = 149,
+	GAME_SERVER_EDIT_TEXT = 150,
+	GAME_SERVER_EDIT_LIST = 151,
+	GAME_SERVER_SEND_GAME_NEWS = 152,
+	GAME_SERVER_SEND_BLESS_DIALOG = 155,
+	GAME_SERVER_BLESSINGS = 156,
+	GAME_SERVER_PRESET = 157,
+	GAME_SERVER_PREMIUM_TRIGGER = 158,
+	GAME_SERVER_PLAYER_DATA_BASIC = 159,
+	GAME_SERVER_PLAYER_DATA = 160,
+	GAME_SERVER_PLAYER_SKILLS = 161,
+	GAME_SERVER_PLAYER_STATE = 162,
+	GAME_SERVER_CLEAR_TARGET = 163,
+	GAME_SERVER_SPELL_DELAY = 164,
+	GAME_SERVER_SPELL_GROUP_DELAY = 165,
+	GAME_SERVER_MULTI_USE_DELAY = 166,
+	GAME_SERVER_PLAYER_MODES = 167,
+	GAME_SERVER_SET_STORE_DEEP_LINK = 168,
+	GAME_SERVER_SEND_RESTING_AREA_STATE = 169,
+	GAME_SERVER_TALK = 170,
+	GAME_SERVER_CHANNELS = 171,
+	GAME_SERVER_OPEN_CHANNEL = 172,
+	GAME_SERVER_OPEN_PRIVATE_CHANNEL = 173,
+	GAME_SERVER_RULE_VIOLATION_CHANNEL = 174,
+	GAME_SERVER_RULE_VIOLATION_REMOVE = 175,
+	GAME_SERVER_EXPERIENCE_TRACKER = 175,
+	GAME_SERVER_RULE_VIOLATION_CANCEL = 176,
+	GAME_SERVER_RULE_VIOLATION_LOCK = 177,
```

Comment:
Violations need to be mapped?

`GAME_SERVER_EXPERIENCE_TRACKER` = `GAME_SERVER_RULE_VIOLATION_REMOVE`